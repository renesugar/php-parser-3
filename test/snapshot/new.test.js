const parser = require('../main');

describe("new", function() {
  it("simple", function() {
    expect(parser.parseEval('new Foo();')).toMatchSnapshot();
  });
  it("variable", function() {
    expect(parser.parseEval('new $var;')).toMatchSnapshot();
  });
  it("simple (2)", function() {
    expect(parser.parseEval('new \\Foo();')).toMatchSnapshot();
  });
  it("simple (3)", function() {
    expect(parser.parseEval('new Foo\\Foo();')).toMatchSnapshot();
  });
  it("simple (4)", function() {
    expect(parser.parseEval('new \\Foo\\Foo();')).toMatchSnapshot();
  });
  it("no parens", function() {
    expect(parser.parseEval('new Foo;')).toMatchSnapshot();
  });
  it("anonymous", function() {
    expect(parser.parseEval('new class() {};')).toMatchSnapshot();
  });
  it("anonymous no parens", function() {
    expect(parser.parseEval('new class {};')).toMatchSnapshot();
  });
  it("anonymous with argument", function() {
    expect(parser.parseEval('new class($var) {};')).toMatchSnapshot();
  });
  it("anonymous with multiple argument", function() {
    expect(parser.parseEval('new class($one, $two, $three) {};')).toMatchSnapshot();
  });
  it("parent", function() {
    expect(parser.parseEval('new parent();')).toMatchSnapshot();
  });
  it("self", function() {
    expect(parser.parseEval('new self();')).toMatchSnapshot();
  });
  it("static", function() {
    expect(parser.parseEval('new static();')).toMatchSnapshot();
  });
});
