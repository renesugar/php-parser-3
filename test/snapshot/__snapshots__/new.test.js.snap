// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new anonymous 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [],
        "kind": "new",
        "what": Class {
          "body": Array [],
          "extends": null,
          "implements": null,
          "isAbstract": false,
          "isAnonymous": true,
          "isFinal": false,
          "kind": "class",
          "name": null,
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new anonymous no parens 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [],
        "kind": "new",
        "what": Class {
          "body": Array [],
          "extends": null,
          "implements": null,
          "isAbstract": false,
          "isAnonymous": true,
          "isFinal": false,
          "kind": "class",
          "name": null,
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new anonymous with argument 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [
          Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "var",
          },
        ],
        "kind": "new",
        "what": Class {
          "body": Array [],
          "extends": null,
          "implements": null,
          "isAbstract": false,
          "isAnonymous": true,
          "isFinal": false,
          "kind": "class",
          "name": null,
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new anonymous with multiple argument 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [
          Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "one",
          },
          Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "two",
          },
          Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "three",
          },
        ],
        "kind": "new",
        "what": Class {
          "body": Array [],
          "extends": null,
          "implements": null,
          "isAbstract": false,
          "isAnonymous": true,
          "isFinal": false,
          "kind": "class",
          "name": null,
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new no parens 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [],
        "kind": "new",
        "what": ClassReference {
          "kind": "classreference",
          "name": "Foo",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new parent 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [],
        "kind": "new",
        "what": ParentReference {
          "kind": "parentreference",
          "raw": "parent",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new self 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [],
        "kind": "new",
        "what": SelfReference {
          "kind": "selfreference",
          "raw": "self",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new simple (2) 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [],
        "kind": "new",
        "what": ClassReference {
          "kind": "classreference",
          "name": "\\\\Foo",
          "resolution": "fqn",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new simple (3) 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [],
        "kind": "new",
        "what": ClassReference {
          "kind": "classreference",
          "name": "Foo\\\\Foo",
          "resolution": "qn",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new simple (4) 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [],
        "kind": "new",
        "what": ClassReference {
          "kind": "classreference",
          "name": "\\\\Foo\\\\Foo",
          "resolution": "fqn",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new simple 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [],
        "kind": "new",
        "what": ClassReference {
          "kind": "classreference",
          "name": "Foo",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new static 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [],
        "kind": "new",
        "what": StaticReference {
          "kind": "staticreference",
          "raw": "static",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`new variable 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": New {
        "arguments": Array [],
        "kind": "new",
        "what": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "var",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
